### PR TITLE
style(ui): Merge requests

### PR DESCRIPTION
**Pull request by juliendonck**
@brandonhaslegs and I styled all the things mapped out in [this document](https://www.notion.so/monadic/2021-03-11-Project-Review-40722756aef449a79af5a60438679d2c)

- [x] Tabs to filter merge requests (closed, open, my own)
- [x]  Empty merge request screen
- [x]  `MergeRequestCard`
- [x]  Header in `MergeRequest`
- [x]  Modals to show terminal instructions `CheckoutMergeRequestButton` `AcceptMergeRequestButton` `MergeRequestButton`

### CODE REVIEW

**Review by geigerzaehler — COMMENTED** [review]
This looks very nice!

What is missing are empty screens when there are no merge requests. We need this for every selected tab basically. It’s ok for me if we address this as a follow-up though.

File: ui/Screen/Project/Source/CheckoutMergeRequestButton.svelte [hunk 1]
```diff
@@ -49,18 +44,16 @@
...
-          class="typo-text-small-mono"
-          style="color: var(--color-foreground-level-6); overflow: scroll">
-          {instructions}
-        </p>
-      </Copyable>
-    </Hoverable>
+    <Copyable bind:this={copyable} showIcon={false}>
+      <p
+        class="typo-text-small-mono"
+        style="color: var(--color-foreground-level-6); overflow-x: scroll; padding: .5rem .5rem .5rem .25rem">
```
Comment (brandonhaslegs) [inline ui/Screen/Project/Source/CheckoutMergeRequestButton.svelte:50]: Does this need a ending semicolon?

File: ui/Screen/Project/Source/AcceptMergeRequestButton.svelte [hunk 2]
```diff
@@ -48,18 +43,16 @@
     <p style="margin-bottom: 0.5rem;">
       To merge this merge request, run this in your terminal:
     </p>
-    <Hoverable bind:hovering={hover}>
-      <Copyable bind:this={copyable} showIcon={hover}>
-        <p
-          class="typo-text-small-mono"
-          style="color: var(--color-foreground-level-6); overflow: scroll">
```
Comment (brandonhaslegs) [inline ui/Screen/Project/Source/AcceptMergeRequestButton.svelte:55]: Semicolon

File: ui/DesignSystem/Component/SegmentedControl.svelte [hunk 3]
```diff
@@ -27,13 +27,16 @@
...
+    border: 1px solid var(--color-foreground-level-3);
+    border-radius: 0.25rem;
   }
   .segmented-control:hover button.active:not(:hover) {
     background: none;
   }
   .segmented-control button {
     cursor: pointer;
-    padding: 8px 16px;
+    padding: 9px 16px;
```
Comment (brandonhaslegs) [inline ui/DesignSystem/Component/SegmentedControl.svelte:39]: Should be rems?

**Comment by juliendonck** [comment]
> This looks very nice!
> 
> 
> 
> What is missing are empty screens when there are no merge requests. We need this for every selected tab basically. It’s ok for me if we address this as a follow-up though.

What do you mean 'every selected tab'? We made an emptystate for the mergerequest page. Source and commits already have it.

File: ui/DesignSystem/Component/SegmentedControl.svelte (hunk 3 above)
Comment (juliendonck) [reply ui/DesignSystem/Component/SegmentedControl.svelte]: https://github.com/radicle-dev/radicle-upstream/pull/1683/commits/7ab714b2ad576552d8a6fcf77ee9f33f5b5641a3

File: ui/Screen/Project/Source/CheckoutMergeRequestButton.svelte (hunk 1 above)
Comment (juliendonck) [reply ui/Screen/Project/Source/CheckoutMergeRequestButton.svelte]: https://github.com/radicle-dev/radicle-upstream/pull/1683/commits/7ab714b2ad576552d8a6fcf77ee9f33f5b5641a3

File: ui/Screen/Project/Source/MergeRequests.svelte [hunk 4]
```diff
@@ -1,31 +1,55 @@
...
+  import { Button } from "../../../DesignSystem/Primitive";
   import MergeRequestList from "./MergeRequestList.svelte";
 
-  const project: Project = getContext("project-page").project;
-  const select = ({ detail: mergeRequest }: { detail: MergeRequest }) => {
-    push(path.projectSourceMergeRequest(project.urn, mergeRequest));
+  const onSelect = ({ detail: mergeRequest }: { detail: MergeRequest }) => {
+    selectMergeRequest(mergeRequest);
+  };
+  let copyable: string;
```
[the remark this answers is not in the document]
Comment (juliendonck) [reply ui/Screen/Project/Source/MergeRequests.svelte]: https://github.com/radicle-dev/radicle-upstream/pull/1683/commits/7ab714b2ad576552d8a6fcf77ee9f33f5b5641a3

File: ui/Screen/Project/Source/MergeRequests.svelte [hunk 5]
```diff
@@ -1,31 +1,55 @@
...
+        <p
+          class="typo-text-small-mono"
+          style="text-align: left; color: var(--color-foreground-level-6); overflow-x: scroll; padding: .5rem .5rem .5rem .25rem">
+          {instructions}
+        </p>
+      </Copyable>
+      <Button
+        variant="primary"
+        style="display: block; margin: 1rem auto 0;"
+        on:click={copy}>
```
[the remark this answers is not in the document]
Comment (juliendonck) [reply ui/Screen/Project/Source/MergeRequests.svelte]: https://github.com/radicle-dev/radicle-upstream/pull/1683/commits/7ab714b2ad576552d8a6fcf77ee9f33f5b5641a3

File: ui/Screen/Project/Source/MergeRequestCard.svelte [hunk 6]
```diff
@@ -50,49 +49,44 @@
...
+            </p>
+            {#if mergeRequest.identity}
+              <Avatar
+                avatarFallback={mergeRequest.identity.avatarFallback}
+                size="small"
+                style="display: flex; justify-content: flex-start; margin-left: 0.5rem;"
+                title={mergeRequest.identity.metadata.handle}
+                variant="circle" />
+            {:else}
+              <p style="margin-left: 0.5rem;">{mergeRequest.peer_id}</p>
```
[the remark this answers is not in the document]
Comment (juliendonck) [reply ui/Screen/Project/Source/MergeRequestCard.svelte]: https://github.com/radicle-dev/radicle-upstream/pull/1683/commits/7ab714b2ad576552d8a6fcf77ee9f33f5b5641a3

File: ui/Screen/Project/Source/MergeRequestCard.svelte [hunk 7]
```diff
@@ -1,29 +1,28 @@
...
-  const session = getContext("session") as UnsealedSession;
+  export let defaultBranch: Branch;
 
   const mergeInfo = mergeRequest && mergeRequest.merged ? "Closed" : "Opened";
-  const showCheckout = isMaintainer(
-    session.identity.urn,
-    getContext("project-page").project
-  );
+
+  console.log({ mergeRequest });
```
[the remark this answers is not in the document]
Comment (juliendonck) [reply ui/Screen/Project/Source/MergeRequestCard.svelte]: https://github.com/radicle-dev/radicle-upstream/pull/1683/commits/7ab714b2ad576552d8a6fcf77ee9f33f5b5641a3

File: ui/Screen/Project/Source/MergeRequestCard.svelte [hunk 8]
```diff
@@ -1,29 +1,28 @@
...
   .merge-request-card {
     display: flex;
     max-height: 3.2rem;
+    justify-content: space-between;
+    width: 100%;
+  }
+
+  .left {
+    display: flex;
+    flex-direction: row;
```
[the remark this answers is not in the document]
Comment (juliendonck) [reply ui/Screen/Project/Source/MergeRequestCard.svelte]: https://github.com/radicle-dev/radicle-upstream/pull/1683/commits/7ab714b2ad576552d8a6fcf77ee9f33f5b5641a3

File: ui/Screen/Project/Source/MergeRequest.svelte [hunk 9]
```diff
@@ -37,63 +48,84 @@
...
     </div>
   </Header.Back>
+  {#if mergeRequest.description}
+    <div class="desc">
+      <Markdown content={mergeRequest.description} />
+    </div>
+  {/if}
+  <div class="action-box">
+    <CompareBranches
+      bg="background-color: var(--color-background);"
```
[the remark this answers is not in the document]
Comment (juliendonck) [reply ui/Screen/Project/Source/MergeRequest.svelte]: https://github.com/radicle-dev/radicle-upstream/pull/1683/commits/7ab714b2ad576552d8a6fcf77ee9f33f5b5641a3

**Comment by geigerzaehler** [comment]
> What do you mean 'every selected tab'? We made an emptystate for the mergerequest page. Source and commits already have it.

When there are no merge requests or no open or closed merge requests the list is simply empty. Do we want to put more information there?

Screenshot from 2021-03-19 11-36-34

**Comment by juliendonck** [comment]
> > What do you mean 'every selected tab'? We made an emptystate for the mergerequest page. Source and commits already have it.
> 
> When there are no merge requests or no open or closed merge requests the list is simply empty. Do we want to put more information there?

Resolved: https://github.com/radicle-dev/radicle-upstream/pull/1683/commits/c29d415dbe0d9c7056a52f6a40a4537e894ba699

**Review by geigerzaehler — APPROVED** [review]
:clap: Thanks for the help!

**Pull request merged**